test_scripts/Polices/External_UCS/018_ATF_P_TC_PTU_with_disallowed_by_external_consent_entities_off_struct.lua

```diff
@@ -0,0 +1,129 @@
+-- b. add this “disallowed_by_external_consent_entities_off:
+-- [entityType: <Integer>, entityId: <Integer>]” field with the corresponding value
+-- to the corresponding "<functional grouping>" to the Policies database.
+--
+-- Preconditions:
+-- 1. Start SDL (make sure 'disallowed_by_external_consent_entities_off' section is omitted in PreloadedPT)
+--
+-- Steps:
+-- 1. Register app1
+-- 3. Activate app1
+-- 4. Perform PTU (make sure 'disallowed_by_external_consent_entities_off' section is defined)
+-- 5. Verify status of update
+-- 6. Register app2
+-- 7. Activate app2
+-- 8. Verify PTSnapshot
+--
+-- Expected result:
+-- a. PTU finished successfully with UP_TO_DATE status
+-- b. PTSnapshot contains 'disallowed_by_external_consent_entities_off' section
+--
+-- Note: Script is designed for EXTERNAL_PROPRIETARY flow
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared Libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local testCasesForExternalUCS = require('user_modules/shared_testcases/testCasesForExternalUCS')
+
+--[[ Local variables ]]
+local checkedStatus = "UP_TO_DATE"
+local checkedSection = "disallowed_by_external_consent_entities_off"
+local grpId = "Location-1"
+
+--[[ General Precondition before ATF start ]]
+commonFunctions:SDLForceStop()
+commonSteps:DeleteLogsFileAndPolicyTable()
+testCasesForExternalUCS.removePTS()
+
+--[[ General Settings for configuration ]]
+Test = require("user_modules/connecttest_resumption")
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:ConnectMobile()
+  self:connectMobile()
+end
+
+function Test:StartSession()
+  testCasesForExternalUCS.startSession(self, 1)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+function Test:RAI_1()
+  testCasesForExternalUCS.registerApp(self, 1)
+end
+
+function Test:ActivateApp_1()
+  local updateFunc = function(pts)
+    pts.policy_table.functional_groupings[grpId][checkedSection] = {
+      {
+        entityID = 128,
+        entityType = 0
+      }
+    }
+  end
+  testCasesForExternalUCS.activateApp(self, 1, checkedStatus, updateFunc)
+end
+
+function Test:CheckStatus_UP_TO_DATE()
+  local reqId = self.hmiConnection:SendRequest("SDL.GetStatusUpdate")
+  EXPECT_HMIRESPONSE(reqId, { status = checkedStatus })
+end
+
+function Test.RemovePTS()
+  testCasesForExternalUCS.removePTS()
+end
+
+function Test:StartSession()
+  testCasesForExternalUCS.startSession(self, 2)
+end
+
+function Test:RAI_2()
+  testCasesForExternalUCS.registerApp(self, 2)
+end
+
+function Test:ActivateApp_2()
```

Comment:
@istoimenova, Agree, but in script activation is linked with PTU and required in order to have table from snapshot created